File: Nx3CS/bin/test/src/nx3/PComplexTieTargetCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PComplexTieTargetCalculator : global::haxe.lang.HxObject {
		public    PComplexTieTargetCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PComplexTieTargetCalculator(global::Array<object> tieinfos){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				global::nx3.PComplexTieTargetCalculator.__hx_ctor_nx3_PComplexTieTargetCalculator(this, tieinfos);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PComplexTieTargetCalculator(global::nx3.PComplexTieTargetCalculator __temp_me98, global::Array<object> tieinfos){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				__temp_me98.tieinfos = tieinfos;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				return new global::nx3.PComplexTieTargetCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				return new global::nx3.PComplexTieTargetCalculator(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ));
			}
			#line default
		}
		
		
		public  global::Array<object> tieinfos;
		
		public virtual   global::Array<object> findTargetHeads(){
			unchecked {
				#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				{
					#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
					int _g = 0;
					#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
					global::Array<object> _g1 = this.tieinfos;
					#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
					while (( _g < _g1.length )){
						#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						object info = _g1[_g];
						#line 20 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						 ++ _g;
						#line 22 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						global::nx3.PHead head = ((global::nx3.PHead) (global::haxe.lang.Runtime.getField(info, "head", 1158359328, true)) );
						int headlevel = head.nhead.level;
						global::nx3.PNote nextnote = head.getNote().getNext();
						if (( nextnote == default(global::nx3.PNote) )) {
							#line 25 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
							continue;
						}
						
						#line 27 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						global::Array<object> nextheads = nextnote.getHeads();
						#line 29 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						{
							#line 29 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
							int _g2 = 0;
							#line 29 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
							while (( _g2 < nextheads.length )){
								#line 29 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
								global::nx3.PHead nexthead = ((global::nx3.PHead) (nextheads[_g2]) );
								#line 29 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
								 ++ _g2;
								#line 31 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
								if (( headlevel == nexthead.nhead.level )) {
									#line 33 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
									global::haxe.lang.Runtime.setField(info, "target", 116192081, nexthead);
									break;
								}
								
							}
							
						}
						
					}
					
				}
				
				#line 38 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				return this.tieinfos;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				switch (hash){
					case 424346869:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						this.tieinfos = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				switch (hash){
					case 1981340297:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("findTargetHeads") ), ((int) (1981340297) ))) );
					}
					
					
					case 424346869:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return this.tieinfos;
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				switch (hash){
					case 1981340297:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return this.findTargetHeads();
					}
					
					
					default:
					{
						#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				baseArr.push("tieinfos");
				#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
				{
					#line 9 "F:\\nx3\\src\\nx3\\PComplexTieTargetCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/PComplexTieTargetCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PComplexTieTargetCalculator extends haxe.lang.HxObject
{
	public    PComplexTieTargetCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PComplexTieTargetCalculator(haxe.root.Array<java.lang.Object> tieinfos)
	{
		nx3.PComplexTieTargetCalculator.__hx_ctor_nx3_PComplexTieTargetCalculator(this, tieinfos);
	}
	
	
	public static   void __hx_ctor_nx3_PComplexTieTargetCalculator(nx3.PComplexTieTargetCalculator __temp_me93, haxe.root.Array<java.lang.Object> tieinfos)
	{
		__temp_me93.tieinfos = tieinfos;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PComplexTieTargetCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PComplexTieTargetCalculator(((haxe.root.Array<java.lang.Object>) (arr.__get(0)) ));
	}
	
	
	public  haxe.root.Array<java.lang.Object> tieinfos;
	
	public   haxe.root.Array<java.lang.Object> findTargetHeads()
	{
		{
			int _g = 0;
			haxe.root.Array<java.lang.Object> _g1 = this.tieinfos;
			while (( _g < _g1.length ))
			{
				java.lang.Object info = _g1.__get(_g);
				 ++ _g;
				nx3.PHead head = ((nx3.PHead) (haxe.lang.Runtime.getField(info, "head", true)) );
				int headlevel = head.nhead.level;
				nx3.PNote nextnote = head.getNote().getNext();
				if (( nextnote == null )) 
				{
					continue;
				}
				
				haxe.root.Array<nx3.PHead> nextheads = nextnote.get_heads();
				{
					int _g2 = 0;
					while (( _g2 < nextheads.length ))
					{
						nx3.PHead nexthead = nextheads.__get(_g2);
						 ++ _g2;
						if (( headlevel == nexthead.nhead.level )) 
						{
							haxe.lang.Runtime.setField(info, "target", nexthead);
							break;
						}
						
					}
					
				}
				
			}
			
		}
		
		return this.tieinfos;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef509 = true;
			switch (field.hashCode())
			{
				case 1993034485:
				{
					if (field.equals("tieinfos")) 
					{
						__temp_executeDef509 = false;
						this.tieinfos = ((haxe.root.Array<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef509) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef510 = true;
			switch (field.hashCode())
			{
				case -722209783:
				{
					if (field.equals("findTargetHeads")) 
					{
						__temp_executeDef510 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("findTargetHeads"))) );
					}
					
					break;
				}
				
				
				case 1993034485:
				{
					if (field.equals("tieinfos")) 
					{
						__temp_executeDef510 = false;
						return this.tieinfos;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef510) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef511 = true;
			switch (field.hashCode())
			{
				case -722209783:
				{
					if (field.equals("findTargetHeads")) 
					{
						__temp_executeDef511 = false;
						return this.findTargetHeads();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef511) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("tieinfos");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


